app/js/model/list_model.js

```diff
@@ -48,12 +48,45 @@ export default class ListModel extends Model {
     return detail;
   }
 
+  fillAppDetailsFromEntryPoints(app, appName) {
+    var detail = Object.create(null);
+    detail.manifestURL = app.manifestURL;
+    detail.description = app.manifest.description;
+    detail.author = app.manifest.developer ? app.manifest.developer.name : '';
+    detail.app = app;
+    if (appName == 'contacts') {
```

Comment:
Is there a way we can do this dynamically by iterating the `manifest.entry_points`?